File: www/src/lib/models/status.ts
import type { Project } from './project';

export interface Status {
    id: string;
    name: string;
    category: StatusCategory;
    color: string;
    rank: string;
    description?: string;
    projectId: string;
    project: Project;
}

export enum StatusCategory {
    Proposed = 'proposed',
    Ready = 'ready',
    Active = 'active',
    Paused = 'paused',
    Review = 'review',
    Completed = 'completed',
    Canceled = 'canceled',
    Archived = 'archived',
}

File: api/WebApp.Domain/Entities/Status.cs
namespace WebApp.Domain.Entities;

public sealed record Status
{
    public StatusId Id { get; init; }
    public string Name { get; init; } = null!;
    public StatusCategory Category { get; init; }
    public string Color { get; init; } = null!;
    public string Rank { get; init; } = null!;
    public string? Description { get; init; }
}
